presto-function-namespace-managers/src/main/java/com/facebook/presto/sqlfunction/db/FunctionNamespaceDao.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sqlfunction.db;
+
+import com.facebook.presto.spi.function.QualifiedFunctionName;
+import com.facebook.presto.spi.function.RoutineCharacteristics.Language;
+import com.facebook.presto.spi.function.SqlFunctionId;
+import com.facebook.presto.spi.function.SqlInvokedFunction;
+import com.facebook.presto.spi.function.SqlParameter;
+import com.facebook.presto.spi.type.TypeSignature;
+import org.jdbi.v3.sqlobject.config.RegisterArgumentFactories;
+import org.jdbi.v3.sqlobject.config.RegisterArgumentFactory;
+import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.customizer.Define;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.util.List;
+import java.util.Optional;
+
+@RegisterRowMapper(SqlInvokedFunctionRowMapper.class)
+@RegisterArgumentFactories({
+        @RegisterArgumentFactory(QualifiedFunctionNameArgumentFactory.class),
+        @RegisterArgumentFactory(SqlFunctionIdArgumentFactory.class),
+        @RegisterArgumentFactory(SqlParametersArgumentFactory.class),
+        @RegisterArgumentFactory(TypeSignatureArgumentFactory.class),
+})
+public interface FunctionNamespaceDao
+{
+    @SqlUpdate("CREATE TABLE `<table_name>` (\n" +
+            "  `id` bigint(20) NOT NULL AUTO_INCREMENT,\n" +
+            "  `function_id` varchar(700) NOT NULL,\n" +
+            "  `version` bigint(20) unsigned NOT NULL,\n" +
+            "  `function_name` varchar(512) NOT NULL,\n" +
+            "  `parameters` varchar(2048) NOT NULL,\n" +
+            "  `return_type` varchar(128) NOT NULL,\n" +
+            "  `language` varchar(64) NOT NULL,\n" +
+            "  `deterministic` tinyint(1) NOT NULL,\n" +
+            "  `called_on_null_input` tinyint(1) NOT NULL,\n" +
+            "  `body` mediumtext,\n" +
+            "  `description` text,\n" +
+            "  `created_time` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,\n" +
+            "  `modified_time` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,\n" +
+            "  PRIMARY KEY (`id`),\n" +
+            "  UNIQUE KEY `function_id_version` (`function_id`,`version`),\n" +
```

Comment:
Semantically `function_id` should be a function name + parameters types. However this is not enforced on the tables. So if function_id does not have the same uniqueness as function name + parameters this implementation is in trouble.

Reply:
Removed for now.